src/reactive-elements.js

```diff
@@ -33,9 +33,12 @@ function reactiveElements(elementName, ReactComponent, options) {
       if (useShadowDom) self.attachShadow({ mode: 'open' });
 
       const observer = new MutationObserver(() => {
-        ReactDOM.unmountComponentAtNode(getRenderRoot(self, useShadowDom));
         const props = utils.getProps(self);
-        props.children = utils.getChildren(self);
+        // we reuse the `childrenFragment`, rather than creating a new fragment,
+        // as this will cause a mutation in the current React component (because
+        // of the way that DOM nodes must be unique)
+        // but the fragment is a reference, so this is fine to keep a hold of
+        props.children = this.childrenFragment;
```

Comment:
The mutation observer also watches for changes to the slot/`children`/`subtree`.
Caching the `childrenFragment` will prevent updates from the slot from being rendered.